starknet_py/tests/e2e/account/account_test.py

```diff
@@ -328,3 +331,88 @@ async def test_deploy_account_uses_custom_calldata(
     tx = await client.get_transaction(deploy_result.hash)
     assert isinstance(tx, DeployAccountTransaction)
     assert tx.constructor_calldata == calldata
+
+
+@pytest.mark.asyncio
+async def test_sign_invoke_tx_for_fee_estimation(account, map_contract):
+    call = map_contract.functions["put"].prepare(key=40, value=50)
+    transaction = await account.sign_invoke_transaction(calls=call, max_fee=MAX_FEE)
+
+    signed_transaction = await account.sign_for_fee_estimate(transaction)
+
+    estimation = await account.client.estimate_fee(signed_transaction)
+    assert estimation.overall_fee > 0
+
+    # Verify that the transaction signed for fee estimation cannot be sent
+    with pytest.raises(ClientError):
+        await account.client.send_transaction(signed_transaction)
+
+    # Verify that original transaction can be sent
+    result = await account.client.send_transaction(transaction)
+    await account.client.wait_for_tx(result.transaction_hash)
+
+
+@pytest.mark.asyncio
+async def test_sign_declare_tx_for_fee_estimation(account, map_compiled_contract):
+    transaction = await account.sign_declare_transaction(
+        compiled_contract=map_compiled_contract, max_fee=MAX_FEE
+    )
+
+    signed_transaction = await account.sign_for_fee_estimate(transaction)
+
+    estimation = await account.client.estimate_fee(signed_transaction)
+    assert estimation.overall_fee > 0
+
+    # Verify that the transaction signed for fee estimation cannot be sent
+    with pytest.raises(ClientError):
+        await account.client.declare(signed_transaction)
+
+    # Verify that original transaction can be sent
+    result = await account.client.declare(transaction)
+    await account.client.wait_for_tx(result.transaction_hash)
+
+
+@pytest.mark.parametrize("account_type", ("base", "proxy"))
+@pytest.mark.asyncio
+async def test_sign_deploy_account_tx_for_fee_estimation(
+    client, deploy_account_details_factory, account_type
+):
+    address, key_pair, salt, class_hash = await deploy_account_details_factory.get()
+
+    if account_type == "base":
+        account = Account(
+            address=address,
+            client=client,
+            key_pair=key_pair,
+            chain=StarknetChainId.TESTNET,
+        )
+    else:
+        account = AccountProxy(
+            AccountClient(
+                address=address,
+                client=client,
+                key_pair=key_pair,
+                chain=StarknetChainId.TESTNET,
+                supported_tx_version=1,
+            )
+        )
```

Comment:
I agree its not very good. Defining a custom fixture for single tests seemed overkill though. This could be split into two tests, but we would have a bit of duplication though.

Reply:
But the fixture could be defined above the test. Or a helper function? to clarify the test :)

Reply:
Added a helper function.